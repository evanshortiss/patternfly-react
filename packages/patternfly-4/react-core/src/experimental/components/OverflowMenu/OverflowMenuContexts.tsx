import * as React from 'react';

export const OverflowMenuContext = React.createContext<{
  breakpoint?: 'md' | 'lg' | 'xl'
}>({
  breakpoint: null
});

export const OverflowMenuContentContext = React.createContext<{
  isHidden: boolean
}>({
  isHidden: false
});
