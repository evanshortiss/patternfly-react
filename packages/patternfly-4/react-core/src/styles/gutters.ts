import { getModifier } from '@patternfly/react-styles';

export const GutterSize = {
  sm: 'sm',
  md: 'md',
  lg: 'lg'
};

/**
 * @param styleObj
 * @param size
 * @param defaultValue
 */
export function getGutterModifier(styleObj, size, defaultValue) {
  return getModifier(styleObj, `gutter-${size}`, defaultValue);
}
