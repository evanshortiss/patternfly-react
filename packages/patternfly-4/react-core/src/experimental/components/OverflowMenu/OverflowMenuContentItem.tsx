import * as React from 'react';
import { css } from '@patternfly/react-styles';
import styles from '@patternfly/react-styles/css/components/OverflowMenu/overflow-menu';
import { OverflowMenuContentContext } from './OverflowMenuContexts';

export interface OverflowMenuContentItemProps extends React.HTMLProps<HTMLDivElement> {
  /** Any elements that can be rendered in the menu */
  children?: any;
  /** Additional classes added to the OverflowMenuItem */
  className?: string;
  /** Modifies the overflow menu item visibility */
  isPersistent?: boolean;
}

export const OverflowMenuContentItem: React.SFC<OverflowMenuContentItemProps> = ({
  className,
  children,
  isPersistent = false,
  ...props
}) => (
  <OverflowMenuContentContext.Consumer>
    {value => (
      <div hidden={!isPersistent && value.isHidden} className={css(
        styles.overflowMenuItem,
        isPersistent && styles.modifiers.persistent,
        className
      )}>
        {children}
      </div>
    )}
  </OverflowMenuContentContext.Consumer>
);
