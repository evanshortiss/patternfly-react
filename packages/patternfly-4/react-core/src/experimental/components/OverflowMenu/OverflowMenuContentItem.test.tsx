import * as React from 'react';
import { mount } from 'enzyme';
import styles from '@patternfly/react-styles/css/components/OverflowMenu/overflow-menu';
import { OverflowMenuContentItem } from './OverflowMenuContentItem';
import { OverflowMenuContentContext } from './OverflowMenuContexts';

describe('OverflowMenuContentItem', () => {
  test('isPersistent and Hidden should still show', () => {
    const view = mount(
      <OverflowMenuContentContext.Provider value={{ isHidden: true }}>
        <OverflowMenuContentItem isPersistent />
      </OverflowMenuContentContext.Provider>
    );
    expect(view.find(`.${styles.overflowMenuItem}`).length).toBe(1);
    expect(view.find(`.${styles.modifiers.persistent}`).length).toBe(1);
    expect(view.find({ hidden: true }).length).toBe(0);
    expect(view).toMatchSnapshot();
  });

  test('Hidden but not isPersistent should be hidden', () => {
    const view = mount(
      <OverflowMenuContentContext.Provider value={{ isHidden: true }}>
        <OverflowMenuContentItem />
      </OverflowMenuContentContext.Provider>
    );
    expect(view.find(`.${styles.modifiers.persistent}`).length).toBe(0);
    expect(view.find({ hidden: true }).length).toBe(1);
    expect(view).toMatchSnapshot();
  });
});

