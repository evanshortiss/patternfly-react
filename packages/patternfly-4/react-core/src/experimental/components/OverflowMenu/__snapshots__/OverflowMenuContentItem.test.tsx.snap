// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OverflowMenuContentItem Hidden but not isPersistent should be hidden 1`] = `
<Component>
  <div
    className="pf-c-overflow-menu__item"
    hidden={true}
  />
</Component>
`;

exports[`OverflowMenuContentItem isPersistent and Hidden should still show 1`] = `
<Component
  isPersistent={true}
>
  <div
    className="pf-c-overflow-menu__item pf-m-persistent"
    hidden={false}
  />
</Component>
`;
